lib/global.g

```diff
@@ -112,20 +113,66 @@ MAKE_READ_ONLY_GLOBAL("REREADING");
 ##
   
 BIND_GLOBAL := function( name, val)
-    if not REREADING and ISBOUND_GLOBAL( name ) then
+local   pos,  count; 
+    ## ignore if the function is bound and rebinding is permitted 
+    if not ( ISBOUND_GLOBAL( name ) and  
+           ( name in GLOBAL_REBINDING_LIST ) ) then 
+      ## the next 11 lines contain the original code for this function 
+      if not REREADING and ISBOUND_GLOBAL( name ) then
         if (IS_READ_ONLY_GLOBAL(name)) then
-            Error("BIND_GLOBAL: variable `", name, "' must be unbound");
+          Error("BIND_GLOBAL: variable `", name, "' must be unbound");
         else
-            Print("#W BIND_GLOBAL: variable `", name,"' already has a value\n");
+          Print("#W BIND_GLOBAL: variable `", name,"' already has a value\n");
         fi;
-    fi;
-    ASS_GVAR(name, val);
-    MAKE_READ_ONLY_GLOBAL(name);
-    return val;
+      fi;
+      ASS_GVAR(name, val);
+      MAKE_READ_ONLY_GLOBAL(name);
+      return val;
+    fi; 
+    if ( name in GLOBAL_REBINDING_LIST ) then 
+      ## increment the count for 'name' 
+      pos := POS_LIST_DEFAULT( GLOBAL_REBINDING_LIST, name, 0 );   
+      count := GLOBAL_REBINDING_COUNT[pos] + 1; 
+      GLOBAL_REBINDING_COUNT[pos] := count; 
+    fi; 
 end;
 
 #############################################################################
 ##
+#F  AllowGlobalRebinding( <list> ) . . function(s) may be BIND_GLOBAL'ed twice
+##
+BIND_GLOBAL( "AllowGlobalRebinding", function( arg ) 
+    local  L, pos, name, val;
+    ##  form the arguments into a list L 
+    if LEN_LIST(arg) = 1 then 
+        if IS_STRING_REP( arg[1] ) then 
+            L := arg;
+        elif IS_LIST( arg[1] ) then 
+            L := arg[1]; 
+        else 
+            L := [ ]; 
+        fi;
+    else 
+        L := arg;
+    fi;
+    for name in L do 
+        ##  prevent duplicate entries 
+        pos := POS_LIST_DEFAULT( GLOBAL_REBINDING_LIST, name, 0 ); 
+        if ( ( pos = fail ) and IS_STRING_REP( name ) ) then 
```

Comment:
Don't like the `IS_STRING_REP` here? Erroring is reasonable, but don't just silently ignore them.